[irods/irods#6597] Add Ubuntu 22.04 support

alanking opened the pull request:
Adds externals builder and irods_core_builder Dockerfiles.

---

Companion PRs:
https://github.com/irods/irods/pull/6620
https://github.com/irods/irods_client_icommands/pull/323

alanking commented:
Gonna leave this in draft until #74 goes in and then adapt changes to match the Various Improvements.

alanking commented:
Still need to add a runner, debugger, etc. Willing to delegate those to a separate issue and/or PR. Just confirmed that these work with main of irods/externals. Need to try again with jwt-cpp but I do not anticipate any issues

SwooshyCueb commented:
The new dockerfiles are still using the old syntax. If merged as-is, I'll update them myself

alanking commented:
@SwooshyCueb - Seems fine to me if you are up for it!

alanking commented:
Ah, hold on. I forgot to try building iRODS. :facepalm: Have only confirmed externals. Setting back to draft status.

alanking commented:
I updated the Dockerfile syntax and added support for building remote repos. Confirmed that building externals and building iRODS / iCommands is working.

jammy doesn't have anything in packages.irods.org yet, so we will need to delete the last commit to enable that repo before merging. Is it possible to make an empty / dummy directory for now so we don't need to worry about publishing unsavory packages?

korydraughn commented:
So you're confident that everything for running the iRODS server in ub22 is good?

What's left for this (other than the `DELETEME` commit)? Running unit/core tests?

trel commented:
i pushed jammy packages to core-dev and unstable... can push to packages too if they're behaving.

korydraughn commented:
I vote yes since no one should have an apt repo configured for ub22 externals yet.

trel commented:
jammy externals now on packages.irods.org

alanking commented:
Core tests are running. I experienced a few failures but so far these seem to be due to the spurious failure as reported here: https://github.com/irods/irods/issues/6954

korydraughn commented on README.md:
````diff
@@ -89,6 +94,27 @@ $ docker run --rm \
              irods-core-builder-ubuntu20 --ccache
 ```
 
+Using `--icommands-only` will invalidate `--core-only` and any options which affect iRODS core builds.
````
By **invalidate**, do you mean it will have higher precedence?

Can you give me an example of which options could be affected as it relates to `--core-only`?

Just want to get it locked down in my head.

alanking replied:
`--irods-repo-url`, `--irods-repo-branch`, `--exclude-microservice-tests`, and `--exclude-unit-tests` will mean nothing when `--icommands-only` is used because the iRODS core packages will not be built and these options are related to that build.

korydraughn reviewed (approved):
Very happy to see the ability to clone remote repos landing.

If you're happy with it, proceed with merging.
